Respect orientation when laying out artwork and player controls

sztomek opened the pull request:
## Description
<!-- Please include a summary of what this PR is changing and why these changes are needed. -->
It's been reported that the expanded player won't display player controls while the phone is in landscape.
I traced down the issue to `ArtworkSection` composable, there was a piece of code that incorrectly selected orientation and hence layout configuration. 
The original conditions made sure to select the proper configuration for previews, but since we are running the app on a real device, that code path was never selected.


Fixes #3901 

## Testing Instructions
1. Install and run app on your phone
2. Optional: Connect to headless auto emulator
3. Start playing a podcast episode while your phone is in portrait mode
4. Rotate your phone to landscape (and make sure you've turned off 'lock to portrait' in settings)
- [ ] Player controls appear as expected

## Screenshots or Screencast 
| Before | After |
| --- | --- |
| ![Screenshot_20250421_142531](https://github.com/user-attachments/assets/6a462705-3d7b-4289-a0c2-ffba85002d7b) | ![Screenshot_20250421_153520](https://github.com/user-attachments/assets/d690978c-1813-4ef2-8c9e-e2ab40f01da4) |


## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...
<!-- If this PR does not contain UI changes, ignore these items -->
- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

geekygecko reviewed (commented):
Thanks for looking into this issue. I couldn't manage to repeat the issue on the `main` branch. I found a slight issue with landscape on tablets.

Before
![Before-Tablet](https://github.com/user-attachments/assets/ea0ceaaa-74a3-4072-b312-01e1c270eb95)

After
![After-Tablet](https://github.com/user-attachments/assets/af897851-f4e1-4074-a82a-d052e2adbb3c)

It would be good to also fix the Compose preview:

Before
<img width="451" alt="Screenshot 2025-04-22 at 11 03 59 am" src="https://github.com/user-attachments/assets/10103e98-e9d4-4aa5-a325-c5eeefeeccc8" />

After
<img width="341" alt="Screenshot 2025-04-22 at 11 04 37 am" src="https://github.com/user-attachments/assets/dff4bd15-d095-4330-b833-554bd10bdfa2" />

sztomek commented:
Thanks @geekygecko, nice catch!

I dug a bit deeper on this, and found what caused this oddity.
My Pixel 8 got classified as `WindowHeightSizeClass.Medium` by height. As the [official docs state](https://developer.android.com/develop/ui/compose/layouts/adaptive/use-window-size-classes), screens above 480dp height in landscape mode are treated as `Medium` size devices. The exact dp value in my case is `485dp` :| 
<img width="863" alt="Screenshot 2025-04-22 at 13 40 42" src="https://github.com/user-attachments/assets/938147f8-065f-4d4b-bb42-a3f9f63b6f82" />

I reverted my changes and added a tolerance of 5% to our logic that decides the height class (see commit a7cbaca218619228634f5a560b0ecd2e59e2b57d). Do you think that tolerance level is acceptable?

Tested both tablet and phone in landscape and port orientations, player controls were displayed as expected.
